lib/qmp.ml

```diff
@@ -258,9 +268,19 @@ let message_of_string str =
         Device.VCPU { id; socket_id; core_id; thread_id}
       in
 
+      let device_add_pci json =
+    (* type t = { id: string; bus: string; hostaddr: string; }*)
+        let id = json         |> arguments |> U.member "id"         |> U.to_string in
+        let bus = json        |> arguments |> U.member "bus"        |> U.to_string in
```

Comment:
Isn't this "addr"? (which would be a device:function pair, perhaps we can have a small record `type devfn = { dev: int; fn: int}`